src/Infrastructure.VS/DocumentEvents/ActiveDocumentTracker.cs

```diff
@@ -29,12 +29,13 @@ namespace SonarLint.VisualStudio.Infrastructure.VS.DocumentEvents
     public interface IActiveDocumentTracker : IDisposable
     {
         /// <summary>
-        /// Raises an event when a text document is brought into focus.
+        /// Raises an event when the active text document changes
         /// </summary>
         /// <remarks>
-        /// Returned <see cref="DocumentFocusedEventArgs.TextDocument"/> cannot be null.
```

Comment:
It turns out that this constraint doesn't work - when the last document is closed, the "focused" document will be null.

Also, referring to "focus" isn't strictly accurate. Selecting a tool window changes the focus, so it's possible there isn't a document with the focus.

However, VS will still know which document is "active" i.e. the  document with focus, or the last document that had the focus. So, I renamed the event args class and the method on the interface.

Reply:
I'm confused, when would this happen?
Based on the code, we raise the event only when the document is not null:
https://github.com/SonarSource/sonarlint-visualstudio/blob/f331d67a5894b1f31d14593b3931c7c95ea18e84/src/Infrastructure.VS/DocumentEvents/ActiveDocumentTracker.cs#L72
So the old comment should be correct, no?

Reply:
This PR is prep work to fix the bug, which will involve changing the behaviour so that the "active document" in the event args can be null.